client-participation/gulpfile.js

```diff
@@ -557,7 +557,7 @@ gulp.task("scriptsOther", function () {
 
 gulp.task("scriptsD3v4", function () {
   var files = [];
-  files.push("node_modules/d3-v4/build/d3.min.js");
+  files.push("js/3rdparty/d3.v4.min.js");
```

Comment:
This should be loading from the node_modules normally, was it not? [I've forked d3 v4](https://github.com/sirodoht/compdemocracy-d3-v4) especially for this reason and I have changed it [here](https://github.com/sirodoht/suburb/blob/798efe768921e46fac845a049c276ebd79f95c35/client-participation/package.json#L27).

Reply:
The `d3-v4` NPM package doesn't seem to contain a built artefact (`d3.min.js`) and instead seems to be a developer package - you can go into the `node_modules` folder and build it by hand but this isn't great for CI/CD and was causing the current build process to fail.

The README directs you to use a CDN to get the built artefact but instead I took the built package and followed the pattern for JQuery by having a static vendor file in the codebase. Since this is version pinned (at v4) anyway this seemed like a reasonable solution. 

Personally I'm not a fan of having custom NPM modules for a bunch of reasons but happy to talk about the pros and cons of custom NPM modules vs CDNs vs vendor code.

Reply:
I see. I actually had the same issue with handlebars and I just pushed the artefact in the repo that we load as node_module ([here](https://github.com/sirodoht/compdemocracy-handlebars-v1)). But, yeah I'm definitely in for avoiding a CDN.